### PR TITLE
Add button plugin version 0.0.1

### PR DESCRIPTION
**Pull request by BasharQaddumi**

### CODE REVIEW

File: packages/plugin-button/src/components/color-picker.jsx [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+ColorPicker.propTypes = {
+  theme: PropTypes.object,
+  style: PropTypes.object,
+  children: PropTypes.string,
+  color: PropTypes.string,
+  onClick: PropTypes.func,
+  onChange: PropTypes.func,
+  flag: PropTypes.bool,
+  colors: PropTypes.object,
```
[the remark this answers is not in the document]
Comment (BasharQaddumi) [reply packages/plugin-button/src/components/color-picker.jsx]: Removed

File: packages/common/src/Components/TextInput.jsx [hunk 2]
```diff
@@ -19,10 +19,11 @@ class TextInput extends React.Component {
           className={classNames(styles.textInput_input, { [styles.textInput_input_invalid]: error })}
           {...otherProps}
         />
-        {error &&
-        <Tooltip shouldRebuildOnUpdate={() => !!error} content={error} theme={theme} moveBy={{ y: 0 }} type={'error'}>
-          <ErrorIcon className={styles.textInput_errorIcon} />
-        </Tooltip>
+        {error && (showTooltip ?
```
Comment (FerasAMasood) [inline packages/common/src/Components/TextInput.jsx]: This change will break all the error tooltips in RCE !!
showTooltip will be undefined for all the cases that it was not passed in, and thus they will all have no tooltip or have to go and configure it.
You should a 'true' default value for that prop

File: packages/plugin-button/src/components/color-picker.jsx [hunk 3]
```diff
@@ -138,6 +133,9 @@ class ColorPicker extends PureComponent {
       }
       return (
         <div
+          role="button"
+          tabIndex="0"
+          onKeyPress={this.onPaletteClick.bind(this, color, index)}
```
Comment (BasharQaddumi) [inline packages/plugin-button/src/components/color-picker.jsx]: The function of onPaletteClick should not be in 'onKeyPress'. any key pressed on keyboard will toggle the color picker and cause strange behavior

File: packages/plugin-button/src/components/color-picker.jsx [hunk 4]
```diff
@@ -173,6 +171,9 @@ class ColorPicker extends PureComponent {
             <div className={styles.palettes}>
               {palattes}
               <div
+                role="button"
+                tabIndex="0"
+                onKeyPress={this.onPaletteDropperClick.bind(this)}
```
Comment (BasharQaddumi) [inline packages/plugin-button/src/components/color-picker.jsx]: The function onPaletteClick should not be in 'onKeyPress'. any key pressed on keyboard will toggle the color picker and cause strange behavior

File: packages/plugin-button/statics/styles/settings-component-styles.scss [hunk 5]
```diff
@@ -64,6 +64,9 @@
 .section_content {
   padding-right: 30px;
   padding-top: 3px;
+  input[type="text"] {
```
Comment (BasharQaddumi) [inline packages/plugin-button/statics/styles/settings-component-styles.scss]: Don't use nested CSS classes

File: packages/plugin-button/statics/styles/button-viewer.scss [hunk 6]
```diff
@@ -0,0 +1,6 @@
+@import '~wix-rich-content-common/dist/statics/styles/mixins';
```
Comment (bchmn) [inline packages/plugin-button/statics/styles/button-viewer.scss]: there's no point in empty classes here, remove this file and add them to the example app theme instead.

File: packages/plugin-link/src/toolbar/TextLinkPanel.jsx [hunk 7]
```diff
@@ -32,7 +32,7 @@ export default class TextLinkPanel extends Component {
       onOverrideContent: this.props.onOverrideContent,
       uiSettings
     };
-
+
```
Comment (bchmn) [inline packages/plugin-link/src/toolbar/TextLinkPanel.jsx]: revert

**Pull request closed without merging**